pkg/envoy/ads/stream.go

```diff
@@ -293,3 +299,50 @@ func isCNforProxy(proxy *envoy.Proxy, cn certificate.CommonName) bool {
 	identityForCN := identity.K8sServiceAccount{Name: chunks[0], Namespace: chunks[1]}
 	return identityForCN == proxyIdentity
 }
+
+// recordPodMetadata records pod metadata and verifies the certificate issued for this pod
+// is for the same service account as seen on the pod's service account
+func (s *Server) recordPodMetadata(p *envoy.Proxy) error {
```

Comment:
Suggest split recording and verifying into two methods

Reply:
dealing with this on a commit on top